### PR TITLE
Add /api/whoami to Butter [#1234]

### PR DESCRIPTION
**Pull request by jbuck**

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @brettgaylor received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/47d275eea3dc177/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/47d275eea3dc177/output.txt

Total script time: 0.99 mins

Server is running on http://pr328.butyr.org/templates/test.html

File: cornfield/app.js [hunk 1]
```diff
@@ -275,6 +275,21 @@ app.post('/project/:id?', function( req, res ) {
...
+  if ( !email ) {
+    res.json( { error: 'unauthorized' }, 403 );
+    return;
+  }
+
+  res.json({
+    email: email,
+    name: email,
+    username: email
+  });
```
Comment (mjschranz) [inline cornfield/app.js:290]: I'm mostly just curious as to why all of them are assigned email. Probably something I just don't know about our database info.

File: src/cornfield/module.js [hunk 2]
```diff
@@ -45,6 +47,31 @@ define(['util/xhr'], function(XHR) {
...
+            }
+          } catch ( err ) {
+            response = {
+              error: "failed to parse data from server: \n" + this.response
+            };
+          }
+
+          if ( callback ) {
+            callback( response );
+          }
```
Comment (mjschranz) [inline src/cornfield/module.js:70]: Could we adopt this style for all the methods for all the functions (other than login)? 

As it stands for the rest of them the callback is being called automatically on error. I imagine another error might be thrown and cause things to potentially break if they didn't pass in a callback.

Maybe it should be left to another ticket. Just a thought.

Comment (jbuck) [inline src/cornfield/module.js:70]: Yeah, I'll file another ticket on this

File: cornfield/app.js (hunk 1 above)
Comment (jbuck) [inline cornfield/app.js:290]: Yeah, we don't store name or username like Popcorn Gallery does, so I'm just using email for all 3 attributes instead

File: test/io/cornfield.js [hunk 3]
```diff
@@ -23,9 +23,8 @@
 
           module( "Unauthenticated tests" );
 
-          asyncTest("Sync API", 1, function() {
-            equal(butter.cornfield.user(), null, 'Username is ""');
-            start();
+          test("Sync API", 1, function() {
```
Comment (secretrobotron) [inline test/io/cornfield.js]: Is this style consistent with the rest of the file? Style's still a mess, but let's at least try our best.

File: test/io/cornfield.js [hunk 4]
```diff
@@ -62,7 +70,7 @@
 
             var failSafe = setTimeout(function() {
               start();
-            }, 10000);
+            }, 20000);
```
Comment (secretrobotron) [inline test/io/cornfield.js]: Same as above.

File: test/io/cornfield.js (hunk 3 above)
Comment (secretrobotron) [inline test/io/cornfield.js]: Also, "asyncTest", "Sync API"? ... lol!

Comment (jbuck) [inline test/io/cornfield.js]: yeah basically

**Pull request merged**